posthog/api/error_tracking.py

```diff
@@ -0,0 +1,35 @@
+from django.db.models import QuerySet
+from rest_framework import serializers, viewsets
+
+from posthog.api.forbid_destroy_model import ForbidDestroyModel
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.models.error_tracking import ErrorTrackingGroup
+
+
+class ErrorTrackingGroupSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ErrorTrackingGroup
+        fields = [
+            "assignee",
+        ]
+
+    def update(self, instance: ErrorTrackingGroup, validated_data: dict, **kwargs) -> ErrorTrackingGroup:
+        instance.assignee = validated_data["assignee"]
+        instance.save()
+        return instance
+
+
+class ErrorTrackingGroupViewSet(TeamAndOrgViewSetMixin, ForbidDestroyModel, viewsets.ModelViewSet):
+    scope_object = "error_tracking_group"
+    queryset = ErrorTrackingGroup.objects.all()
+    serializer_class = ErrorTrackingGroupSerializer
+
+    def safely_get_object(self, queryset) -> QuerySet:
+        fingerprint = self.kwargs["pk"]
+        groups = ErrorTrackingGroup.filter_fingerprints(queryset=queryset, fingerprints=[fingerprint])
+
+        if groups:
+            return groups.first()
+
+        # lazily creates error group if it does not already exist
+        return ErrorTrackingGroup.objects.create(team=self.team, fingerprint=fingerprint)
```

Comment:
could do `get_or_create`
https://docs.djangoproject.com/en/5.0/ref/models/querysets/#get-or-create

Reply:
Ah, I originally assumed I couldn't do that because we are doing a complex filter lookup in `filter_fingerprints` which would then set the `merged_fingerprints` on the created objects.

But I think we should only ever lookup a group by it's primary fingerprint so we're probably safe doing `ErrorTrackingGroup.objects.get_or_create(fingerprint=fingerprint, team=self.team)`

Will make that change